### ISSUE TITLE
原文fig3梯度相似性计算时，得到head的相似度最高

### ISSUE THREAD

**Issue by liluhu0** [ArchipLab-LinfengZhang/data-augmentation-multi-task#1]
张老师您好，我最近在尝试计算原图像和增强图像**梯度相似性**时遇到问题，得到的**结果是head的梯度相似性最高**。
**以下是我的具体操作：**
**Model：** 
resnet18
**数据集：** 
CIFAR10
**训练方式：** 
将原图和增强图像拼接后输入网络，对原图（您代码中的transform1）和增强图像（您代码中的transform2）的输出分别与target做CE loss，得到loss1、loss2
**每一层获得梯度相似性方法：**
```
# model代指某一层
model.zero_grad(set_to_none=True)
loss1.backward(retain_graph=True)
grad1 = torch.cat(
    [v.grad.reshape((-1))
        for i,(k,v) in enumerate(model.state_dict(keep_vars=True).items())
        if (key.startswith(k)) and (v.requires_grad)], 
    axis = 0)
model.zero_grad(set_to_none=True)
loss2.backward(retain_graph=True)
grad2 = torch.cat(
    [v.grad.reshape((-1))
        for i,(k,v) in enumerate(model.state_dict(keep_vars=True).items())
        if (key.startswith(k)) and (v.requires_grad)], 
    axis = 0)
similarity = float(torch.cosine_similarity(grad1,grad2,axis = 0))
```

**最终head的部分训练过程中得到的相似度如下：**
![image](https://user-images.githubusercontent.com/87222127/227131692-25b9e6f3-4400-4f50-81e0-b4ae97087b74.png)

**请问其中有错误的步骤吗？如果有希望能得到您的指导！**
